content/documentation/pages/5-feature-guides/3-batch/7-java-dsl.md

```diff
@@ -0,0 +1,163 @@
+To create an instance of a `DataFlowTemplate`, you need to provide a `URI` location of the Data Flow Server.
+Spring Boot auto-configuration for `DataFlowTemplate` is also available.
+You can use the properties in [DataFlowClientProperties](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientProperties.java) to configure the connection to the Data Flow server.
+Generally, you should start with the `spring.cloud.dataflow.client.uri` property.
+
+```Java
+URI dataFlowUri = URI.create("http://localhost:9393");
+DataFlowOperations dataFlowOperations = new DataFlowTemplate(dataFlowUri);
+```
+
+## Task DSL Usage
+
+Consider the following example, which creates a new composed task:
+
+```Java
+dataFlowOperations.appRegistryOperations().importFromResource(
+                     "https://dataflow.spring.io/task-maven-latest", true);
+
+Task task = Task.builder(dataflowOperations)
+              .name("myComposedTask")
+              .definition("a: timestamp && b:timestamp")
+              .description("My Composed Task")
+              .create();
+```
+
+The `TaskBuilder` class is returned from the `Task.builder(dataFlowOperations)` method.
+
+The `create` method returns an instance of a `Task` definition that represents a composite task that has been created but not launched.
+It takes a single string for the task definition (same as in the shell).
+
+<!--NOTE-->
+
+In order to create and launch your tasks, you need to make sure that the corresponding apps have been registered in the Data Flow server first.
+Attempting to launch a task that contains an unknown application throws an exception.
+You can register your application by using the `DataFlowTemplate`, as follows:
+
+```java
+dataFlowOperations.appRegistryOperations().importFromResource(
+            "https://dataflow.spring.io/task-maven-latest", true);
+```
+
+<!--END_NOTE-->
+
+Instead of creating a new Task you can use the `TaskBuilder` to retrieve an existing Task instances by name:
+
+```Java
+Optional<Task> task = Task.builder(dataflowOperations).findByName("myTask");
+```
+
+or to list all existing tasks:
+
+```Java
+List<Task> tasks = Task.builder(dataflowOperations).allTasks();
+```
+
+With the `Task` instance, you have methods available to `launch` or `destroy` the task.
+The following example launches the task:
+
+```Java
+long launchId = task.launch();
+
+```
+
+The `launchId` is an unique Task execution identifier for the launched task.
+The `launch` method is overloaded to take a `java.util.Map<String, String>` of launch properties and `java.util.List<String>` of the command line arguments.
+
+The tasks are asynchronously executed. If your use case require to wait on task completion or other task state you can you the java concurrency utils or the `Awaitility` library like this:
```

Comment:
This can be rewritten as:
The tasks are asynchronously executed. If your use case requires you to wait on task completion or other task state you can use the java concurrency utils or the `Awaitility` library like this: